shotover-proxy/src/transforms/tee.rs

```diff
@@ -167,3 +186,83 @@ impl Transform for Tee {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::config::topology::TopicHolder;
+    use crate::transforms::TransformsConfig;
+    use std::collections::HashMap;
+
+    #[tokio::test]
+    async fn test_validate_no_subchain() {
+        let holder = TopicHolder {
+            topics_rx: HashMap::new(),
+            topics_tx: HashMap::new(),
+        };
+
+        let mut config = TeeConfig {
+            behavior: Some(ConsistencyBehavior::Ignore),
+            timeout_micros: None,
+            chain: vec![TransformsConfig::Null],
+            buffer_size: None,
+        };
+        let mut transform = config.get_source(&holder).await.unwrap();
+        let mut result = transform.validate();
+        assert_eq!(result, Vec::<String>::new());
+
+        config = TeeConfig {
+            behavior: Some(ConsistencyBehavior::FailOnMismatch),
+            timeout_micros: None,
+            chain: vec![TransformsConfig::Null],
+            buffer_size: None,
+        };
+        transform = config.get_source(&holder).await.unwrap();
+        result = transform.validate();
+        assert_eq!(result, Vec::<String>::new());
+    }
+
+    #[tokio::test]
+    async fn test_validate_invalid_chain() {
+        let holder = TopicHolder {
+            topics_rx: HashMap::new(),
+            topics_tx: HashMap::new(),
+        };
+
+        let config = TeeConfig {
+            behavior: Some(ConsistencyBehavior::SubchainOnMismatch(vec![
+                TransformsConfig::Null,
+                TransformsConfig::Null,
+            ])),
+            timeout_micros: None,
+            chain: vec![TransformsConfig::Null],
+            buffer_size: None,
+        };
+
+        let transform = config.get_source(&holder).await.unwrap();
+        let result = transform.validate();
+        let expected = vec!["Tee:", "  fail_chain:", "    Terminating transform \"Null\" is not last in chain. Terminating transform must be last in chain."];
```

Comment:
its not called fail_chain anymore, did we miss a rename somewhere?

Reply:
https://github.com/shotover/shotover-proxy/blob/54b0e40956ceb46a62513de3b759751a6d0db17c/shotover-proxy/src/transforms/tee.rs#L45

It was always called fail_chain

Reply:
ah, what I meant was, now that the config is called SubchainOnMismatch we should rename "fail_chain" to something like "mismatch_chain"

Reply:
Oh right I thought you meant the other PR. I'll fix that up